File: App/app/src/main/java/com/github/travelplannerapp/di/AppModule.java
package com.github.travelplannerapp.di;

import android.content.Context;

import com.github.travelplannerapp.App;

import dagger.Module;
import dagger.Provides;

/**
 * This is where you will inject application-wide dependencies.
 */
@Module
public class AppModule {

    @Provides
    Context provideContext(App application) {
        return application.getApplicationContext();
    }
}

File: App/app/src/main/java/com/github/travelplannerapp/di/AppModule.kt
package com.github.travelplannerapp.di

import android.content.Context

import com.github.travelplannerapp.App

import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * This is where you will inject application-wide dependencies.
 */
@Module
class AppModule {

    @Provides
    @Singleton
    internal fun provideContext(application: App): Context = application.applicationContext

}